templates/terraform/resource.erb

```diff
@@ -323,10 +329,17 @@ func resource<%= resource_name -%>Update(d *schema.ResourceData, meta interface{
             return err
         }
 
+<% if object.autogen_async -%>
```

Comment:
See above here^ how we marshal `res` into `op`; all we need to do should be passing in `res` instead of `op` when we've generated out Operation handling.